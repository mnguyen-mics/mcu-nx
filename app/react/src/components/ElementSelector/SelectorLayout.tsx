import * as React from 'react';
import { Button, Layout } from 'antd';
import ActionBar from '../ActionBar';
import McsIcon from '../McsIcon';
import { FormattedMessage } from 'react-intl';
import { EmptyTableView } from '../TableView';

const { Content } = Layout;

export interface SelectorLayoutProps {
  actionBarTitle: string;
  handleAdd: () => void;
  handleClose: () => void;
  disabled: boolean;
  className?: string;
}

export default class SelectorLayout extends React.Component<
  SelectorLayoutProps,
  any
> {
  render() {
    const {
      actionBarTitle,
      handleAdd,
      handleClose,
      disabled,
      children,
      className,
    } = this.props;

    return (
      <Layout>
        <div className="edit-layout ant-layout">
          <ActionBar paths={[{ name: actionBarTitle }]} edition={true}>
            <Button type="primary" className="mcs-primary" onClick={handleAdd}>
              <McsIcon type="plus" />
              <FormattedMessage
                id="table-selector-add-button"
                defaultMessage="Add"
              />
            </Button>
            <McsIcon
              type="close"
              className="close-icon mcs-table-cursor"
              onClick={handleClose}
            />
          </ActionBar>
          <Layout>
            <Content
              className={`mcs-edit-container ${className ? className : ''}`}
            >
              {disabled ? <EmptyTableView iconType="file" /> : children}
            </Content>
          </Layout>
        </div>
      </Layout>
    );
  }
}
