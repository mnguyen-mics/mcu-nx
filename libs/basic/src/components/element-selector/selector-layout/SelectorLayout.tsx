import * as React from 'react';
import { Button, Layout } from 'antd';
import ActionBar from '../../action-bar';
import McsIcon from '../../mcs-icon';
import EmptyTableView from '../../empty-table-view';

const { Content } = Layout;

export interface SelectorLayoutProps {
  actionBarTitle: string;
  handleAdd: () => void;
  handleClose: () => void;
  disabled: boolean;
  className?: string;
  addButtonText: string;
  noElementText: string;
}

export default class SelectorLayout extends React.Component<SelectorLayoutProps, any> {
  render() {
    const {
      actionBarTitle,
      handleAdd,
      handleClose,
      disabled,
      children,
      className,
      addButtonText,
      noElementText,
    } = this.props;
    const prefixCls = 'mcs-selector-layout';

    return (
      <Layout className={prefixCls}>
        <ActionBar pathItems={[actionBarTitle]} edition={true}>
          <Button className='add-button mcs-addButton' onClick={handleAdd}>
            <McsIcon type='plus' />
            {addButtonText}
          </Button>
          <McsIcon type='close' className='close-icon mcs-table-cursor' onClick={handleClose} />
        </ActionBar>
        <Layout>
          <Content className={`mcs-edit-container ${className ? className : ''}`}>
            {disabled ? <EmptyTableView iconType='warning' message={noElementText} /> : children}
          </Content>
        </Layout>
      </Layout>
    );
  }
}
