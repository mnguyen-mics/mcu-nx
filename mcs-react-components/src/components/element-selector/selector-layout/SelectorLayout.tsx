import * as React from 'react';
import { Button, Layout } from 'antd';
import ActionBar from '../../action-bar';
import McsIcon from '../../mcs-icon';
import { FormattedMessage } from 'react-intl';
import EmptyTableView from '../../empty-table-view';

const { Content } = Layout;

export interface SelectorLayoutProps {
  actionBarTitle: string;
  handleAdd: () => void;
  handleClose: () => void;
  disabled: boolean;
  className?: string;
  noElementMessage: FormattedMessage.MessageDescriptor;
}

export default class SelectorLayout extends React.Component<
  SelectorLayoutProps,
  any
> {
  render() {
    const {
      actionBarTitle,
      handleAdd,
      handleClose,
      disabled,
      children,
      className,
      noElementMessage,
    } = this.props;

    return (
      <Layout>
        <div className="edit-layout ant-layout">
          <ActionBar paths={[{ name: actionBarTitle }]} edition={true}>
            <Button
              type="primary"
              className="mcs-primary"
              onClick={handleAdd}
              href=""
            >
              <McsIcon type="plus" />
              <FormattedMessage
                id="components.elementSelector.selectorLayout.actionbar.add.button"
                defaultMessage="Add"
              />
            </Button>
            <McsIcon
              type="close"
              className="close-icon mcs-table-cursor"
              onClick={handleClose}
            />
          </ActionBar>
          <Layout>
            <Content
              className={`mcs-edit-container ${className ? className : ''}`}
            >
              {disabled ? (
                <EmptyTableView
                  iconType="warning"
                  intlMessage={noElementMessage}
                />
              ) : (
                children
              )}
            </Content>
          </Layout>
        </div>
      </Layout>
    );
  }
}
