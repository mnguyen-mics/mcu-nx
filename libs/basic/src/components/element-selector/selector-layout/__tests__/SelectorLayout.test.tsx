import 'jest';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import SelectorLayout, { SelectorLayoutProps } from '../SelectorLayout';

it('renders the SelectorLayout with no element', () => {
  const props: SelectorLayoutProps = {
    actionBarTitle: 'Action Bar Title',
    handleAdd: () => {
      //
    },
    handleClose: () => {
      //
    },
    disabled: true,
    noElementText: 'No data found',
    addButtonText: 'Add',
  };
  const component = TestRenderer.create(<SelectorLayout {...props} />);
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});

it('renders the SelectorLayout with children', () => {
  const props: SelectorLayoutProps = {
    actionBarTitle: 'Action Bar Title',
    handleAdd: () => {
      //
    },
    disabled: false,
    handleClose: () => {
      //
    },
    noElementText: 'No data found',
    addButtonText: 'Add',
  };
  const component = TestRenderer.create(
    <SelectorLayout {...props}>
      <div>Children elements</div>
    </SelectorLayout>,
  );
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
