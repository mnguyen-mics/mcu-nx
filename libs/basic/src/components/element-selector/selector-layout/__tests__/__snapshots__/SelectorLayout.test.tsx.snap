// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the SelectorLayout with children 1`] = `
<section
  className="ant-layout mcs-selector-layout"
>
  <div
    className="ant-row ant-row-space-between ant-row-middle mcs-actionbar-edit  "
    style={Object {}}
  >
    <nav
      className="ant-breadcrumb mcs-breadcrumb"
    >
      <ol>
        <li>
          <span
            className="ant-breadcrumb-link"
          >
            Action Bar Title
          </span>
          <span
            className="ant-breadcrumb-separator"
          >
            <span
              className="mcs-icon "
            >
              <i
                className="mcs-chevron-right"
                style={Object {}}
              />
            </span>
          </span>
        </li>
      </ol>
    </nav>
    <div
      className="left-part-margin"
    >
      <button
        className="ant-btn ant-btn-default add-button mcs-addButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="mcs-icon "
        >
          <i
            className="mcs-plus"
            style={Object {}}
          />
        </span>
        <span>
          Add
        </span>
      </button>
      <span
        className="mcs-icon close-icon mcs-table-cursor"
        onClick={[Function]}
      >
        <i
          className="mcs-close"
          style={Object {}}
        />
      </span>
    </div>
  </div>
  <section
    className="ant-layout"
  >
    <main
      className="ant-layout-content mcs-edit-container "
    >
      <div>
        Children elements
      </div>
    </main>
  </section>
</section>
`;

exports[`renders the SelectorLayout with no element 1`] = `
<section
  className="ant-layout mcs-selector-layout"
>
  <div
    className="ant-row ant-row-space-between ant-row-middle mcs-actionbar-edit  "
    style={Object {}}
  >
    <nav
      className="ant-breadcrumb mcs-breadcrumb"
    >
      <ol>
        <li>
          <span
            className="ant-breadcrumb-link"
          >
            Action Bar Title
          </span>
          <span
            className="ant-breadcrumb-separator"
          >
            <span
              className="mcs-icon "
            >
              <i
                className="mcs-chevron-right"
                style={Object {}}
              />
            </span>
          </span>
        </li>
      </ol>
    </nav>
    <div
      className="left-part-margin"
    >
      <button
        className="ant-btn ant-btn-default add-button mcs-addButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="mcs-icon "
        >
          <i
            className="mcs-plus"
            style={Object {}}
          />
        </span>
        <span>
          Add
        </span>
      </button>
      <span
        className="mcs-icon close-icon mcs-table-cursor"
        onClick={[Function]}
      >
        <i
          className="mcs-close"
          style={Object {}}
        />
      </span>
    </div>
  </div>
  <section
    className="ant-layout"
  >
    <main
      className="ant-layout-content mcs-edit-container "
    >
      <div
        className="mcs-empty-table-view"
      >
        <div
          className="ant-col ant-col-24 mcs-empty-table-view-content mcs-table-view-empty"
          style={Object {}}
        >
          <div
            className="mcs-empty-table-view-logo"
          >
            <span
              className="mcs-icon "
            >
              <i
                className="mcs-warning"
                style={Object {}}
              />
            </span>
          </div>
          No data found
        </div>
      </div>
    </main>
  </section>
</section>
`;
